plugins/org.jkiss.dbeaver.parser/src/org/jkiss/dbeaver/parser/grammar/RegexExpression.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.parser.grammar;
+
+import java.util.regex.Pattern;
+
+/**
+ * Represents terminal part of the text with regular pattern
+ */
+public class RegexExpression extends TerminalExpression {
+    
+    // see https://docs.oracle.com/javase/tutorial/essential/regex/literals.html
+    private static final Pattern _regexEscapeCharsPattern = Pattern.compile("([\\\\\\.\\?\\!\\[\\]\\{\\}\\(\\)\\<\\>\\*\\+\\-\\=\\^\\$\\|])");
```

Comment:
```suggestion
    private static final Pattern _regexEscapeCharsPattern = Pattern.compile("([\\.?!\[\]{}()<>*+\-=^$|])");
```
Most of the backslashes here do not require escaping.

Reply:
All of these characters are regex special symbols, so they should be escaped to be treated as raw characters, and Java string literal requires backslashes to be escaped with backslashes  to represent backslash character for regex escaping.